esm/ssr.js

```diff
@@ -14,7 +14,7 @@ const cloneNode = (current, newNode) => {
       case 1:
         const element = new Element(node.name);
         for (const {name, value} of node.attributes)
-          element.attributes.push({name, value});
+          element.setAttribute(name, value);
```

Comment:
what do you think about this?

```js
const element = new Element(node.name);
if (node.attributes !== EMPTY)
  element.attributes = node.attributes.map(({name, value}) => ({name, value}));
```

Reply:
wouldn't `node.attributes !== EMPTY` always be false? gonna test

Reply:
oh, i'm mixing `element` and `node` together

Reply:
to be honest ... with the current state I think this would be even better:

```js
const element = new Element(node.name);
element.attributes = node.attributes;
```

the thing is ... attributes are not meant to change as SSR won't do runtime things (right now) so maybe we can cut the loop all together

Reply:
just tested, it works as expected but that also makes me think it might haunt me back in the future ... as clone is what we're after, maybe let's just clone and call it a day ...

```js
// new entry
const cloneAttribute = ({name, value}) => ({name, value});

const cloneNode = (current, newNode) => {
  for (const node of current) {
    switch (node.nodeType) {
      case 1:
        const element = new Element(node.name);
        // previous hint
        if (node.attributes !== EMPTY)
          element.attributes = node.attributes.map(cloneAttribute);
        cloneNode(node, element);
        newNode.appendChild(element);
        break;
      case 3:
        newNode.appendChild(new Text(node.data));
        break;
      case 11:
        for (const inner of node)
          cloneNode(inner, newNode);
        break;
    }
  }
  return newNode;
};
```

Reply:
P.S. no need to bring in those comments 😅

Reply:
last, but not least, please change `test/ssr.jsx` *ul* as `<ul test="attribute">` so we know this actually works as expected (I should've done that before).

Reply:
Stepped away from the computer now, but will fix when I'm back.

Reply:
Let me save you time and thanks for your bug and patience 👋